### PR TITLE
Update SmartReshape tests to check tensor and node names (#68694)

### PR DESCRIPTION
**Pull request by SmirnovGrigory**
### Tickets:
 - *68694*

### CODE REVIEW

File: src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp [hunk 1]
```diff
@@ -106,9 +106,8 @@ ngraph::pass::StridedSliceSqueeze::StridedSliceSqueeze() {
             shrink_axis_mask,
             ellipsis_mask);
 
+        replace_output_update_name(slice->output(0), slice->input_value(0));
```
Comment (GlebKazantaev) [inline src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp:109]:
```suggestion
        return replace_output_update_name(slice->output(0), slice->input_value(0));
```

File: src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp [hunk 2]
```diff
@@ -106,9 +106,8 @@ ngraph::pass::StridedSliceSqueeze::StridedSliceSqueeze() {
...
         replace_node(squeeze, new_slice);
-        new_slice->set_friendly_name(slice->get_friendly_name());
-        copy_runtime_info(slice, new_slice);
         return true;
```
Comment (GlebKazantaev) [inline src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp:111]: remove

File: src/tests/functional/inference_engine/transformations/sr_sub_graph_ops.cpp [hunk 3]
```diff
@@ -8,18 +8,33 @@
...
 #include <cpp/ie_cnn_network.h>
 
+#include <ngraph/pass/manager.hpp>
+#include "graph_comparator.hpp"
+
 using namespace ngraph;
 
+namespace {
+
+    void init_unique_names(std::shared_ptr<ngraph::Function> f, const std::shared_ptr<ngraph::pass::UniqueNamesHolder>& unh) {
```
Comment (GlebKazantaev) [inline src/tests/functional/inference_engine/transformations/sr_sub_graph_ops.cpp:18]: Please move all this function to the ngraph_test_utils.hpp

Comment (SmirnovGrigory) [reply src/tests/functional/inference_engine/transformations/sr_sub_graph_ops.cpp]: done

File: src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp (hunk 1 above)
Comment (SmirnovGrigory) [reply src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp]: done

File: src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp (hunk 2 above)
Comment (SmirnovGrigory) [reply src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp]: done

File: src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp [hunk 4]
```diff
@@ -90,3 +90,17 @@ class TransformationTestsF : public  CommonTestUtils::TestsCommon {
     bool m_soft_names_comparison{true};
     bool m_enable_accuracy_check{false};
 };
+
+namespace {
+    void init_unique_names(std::shared_ptr<ngraph::Function> f, const std::shared_ptr<ngraph::pass::UniqueNamesHolder>& unh) {
```
Comment (GlebKazantaev) [inline src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:95]: Please do not use anonymous namespaces in header files. https://rules.sonarsource.com/cpp/RSPEC-1000
Keep functions declaration in hpp and move definition to cpp.

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

File: src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp (hunk 4 above)
Comment (SmirnovGrigory) [reply src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp]: done

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**